File: battle_real_time/lib/battle_real_time/contracts/publishers/battle_actions/terminate_battle_contract.ex
defmodule BattleRealTime.Contracts.Publishers.BattleActions.TerminateBattleContract do
  @moduledoc """
  Validates the terminate_battle payload published to RabbitMQ.
  """
  use BattleRealTime.Contracts.Contract

  @primary_key false
  embedded_schema do
    field(:battle_id, :string)
    field(:reason, :string)
  end

  def changeset(struct, params) do
    struct
    |> cast(params, [:battle_id, :reason])
    |> validate_required([:battle_id, :reason])
  end
end

File: battle_engine/app/contracts/publishers/terminate_battle_contract.rb
# frozen_string_literal: true

module Contracts
  module Publishers
    class TerminateBattleContract < Dry::Validation::Contract
      params do
        required(:battle_id).filled(:string)
        required(:reason).filled(:string)
        optional(:timestamp).maybe(:string)
      end
    end
  end
end
